cli/cmd/generate.go

```diff
@@ -0,0 +1,348 @@
+				mods = append(mods, aws.UseConsolidatedCloudtrail())
+			}
+
+			// Create new struct
+			data := aws.NewTerraform(
+				GenerateAwsCommandState.AwsRegion,
+				GenerateAwsCommandState.Config,
+				GenerateAwsCommandState.Cloudtrail,
+				mods...)
+
+			// Generate
+			hcl, err := data.Generate()
+			cli.StopProgress()
+
+			if err != nil {
+				return errors.Wrap(err, "failed to generate terraform code")
+			}
+
+			// Write-out generated code to location specified
+			location, err := writeHclOutput(hcl, cmd)
+			if err != nil {
+				return errors.Wrap(err, "failed to write terrraform code to disk")
+			}
+
+			// Prompt to execute
+			execute := false
+			err = SurveyQuestionInteractiveOnly(SurveyQuestionWithValidationArgs{
+				Prompt:   &survey.Confirm{Default: execute, Message: QuestionRunTfPlan},
+				Response: &execute,
+			})
+
+			if err != nil {
+				return errors.Wrap(err, "failed to run terraform execution")
+			}
+
+			// Execute
+			locationDir := filepath.Dir(location)
+			if execute {
+				if err := TerraformPlanAndExecute(locationDir); err != nil {
+					return errors.Wrap(err, "failed to run terraform apply")
+				}
+			}
+
+			// Output where code was generated
+			if !execute {
+				provideGuidanceAfterExit(false, false, locationDir, "terraform")
+			}
+
+			return nil
+		},
+		PreRunE: func(cmd *cobra.Command, _ []string) error {
+			// Load any cached inputs if interactive
+			if cli.InteractiveMode() {
+				cachedOptions := &aws.GenerateAwsTfConfigurationArgs{}
+				if ok := cli.ReadCachedAsset("iac-aws-generate-params", &cachedOptions); ok {
+					cli.Log.Debug("loaded previously set values for AWS iac generation")
+				}
+
+				extraState := &AwsGenerateCommandExtraState{}
+				if ok := cli.ReadCachedAsset("extra-state", &extraState); ok {
+					cli.Log.Debug("loaded previously set values for AWS iac generation (extra state)")
+				}
+
+				// Merge cached inputs to current options (current options win)
+				if err := mergo.Merge(GenerateAwsCommandState, cachedOptions); err != nil {
+					return errors.Wrap(err, "failed to load saved options!")
+				}
+				if err := mergo.Merge(GenerateAwsCommandExtraState, extraState); err != nil {
+					return errors.Wrap(err, "failed to load saved options!")
+				}
+			}
+
+			// Parse passed in subaccounts (if any)
+			if GenerateAwsCommandExtraState.AwsSubAccounts != "" {
+				// validate consolidated_cloudtrail is enabled - otherwise this flag cannot be used
+				if ok, _ := cmd.Flags().GetBool("consolidated_cloudtrail"); !ok {
+					return errors.New("aws subaccounts can only be supplied with consolidated cloudtrail enabled")
+				}
+
+				// validate the format of supplied value is correct
+				matchRegEx := `(\w+:\w+(,)?)+`
+				if ok, _ := regexp.MatchString(matchRegEx, GenerateAwsCommandExtraState.AwsSubAccounts); !ok {
+					return errors.New("supplied aws subaccounts in invalid format")
+				}
+
+				awsSubaccounts := []aws.AwsSubAccount{}
```

Comment:
nitpick: Could this var be changed to camelCase? `awsSubaccounts` --> `awsSubAccounts`

Reply:
👍